run.py

```diff
@@ -122,36 +125,32 @@ def config_validator():
         print("Starting server in ~*TEST MODE*~")
         app.run(host='localhost')
     elif args.backup:
-        # To verify if issues.db exist, simply calling os.path.exists is not enough.
-        # Both os.path.exists and os.path.isfile returns true even when file is not there in the path
-        # Every time the script is re run, os.path.exists returns true because of cache.
-        # To avoid this, we open the file in try catch - just to be sure
         if os.path.isfile(os.path.join(os.getcwd(), 'issues.db')):
-            try:
-                with open(os.path.join(os.getcwd(), 'issues.db')) as file:
-                    if not os.path.exists(app.config['BACKUP_DEFAULT_DEST']):
-                        print 'backup db folder does not exist'
-                        os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
-                    else:
-                        print 'backup db folder exists'
-                        # Retain last 3 recent backup files
-                        no_of_backup_files = len(os.listdir(app.config['BACKUP_DEFAULT_DEST']))
-                        if no_of_backup_files > 3:
-                            for idx, val in enumerate(os.listdir(app.config['BACKUP_DEFAULT_DEST'])):
-                                if idx < no_of_backup_files - 2:
-                                    os.remove(app.config['BACKUP_DEFAULT_DEST'] + val)
-                    current_ts = time.time()
-                    time_stamp = datetime.datetime.fromtimestamp(current_ts).strftime('%Y-%m-%d-%H:%M:%S')
-                    issue_backup_db = 'issues_backup_' + time_stamp + '.db'
-                    os.rename(os.getcwd() + '/issues.db', app.config['BACKUP_DEFAULT_DEST'] + issue_backup_db)
-                    try:
-                        os.remove(os.path.join(os.getcwd(), 'issues.db'))
-                    except OSError:
-                        print 'Backup issues.db complete'
-
-            except IOError:
-                sys.exit(FILE_NOT_FOUND_ERROR)
-
+            issue_engine = create_engine('sqlite:///' + os.path.join(
+                app.config['BASE_DIR'], 'issues.db'))
+            issue_session_maker = sessionmaker(autocommit=False,
+                                               autoflush=False,
+                                               bind=issue_engine)
+            issue_db = scoped_session(issue_session_maker)
+            # Take a backup if issues.db has data dump.
+            if issue_db().execute('select count(*) from webcompat_issues').fetchall()[0][0] > 0:
+                if not os.path.exists(app.config['BACKUP_DEFAULT_DEST']):
+                    print 'Backup db folder does not exist'
+                    os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
+                else:
+                    print 'Backup db folder exists'
+                current_ts = time.time()
+                time_stamp = datetime.datetime.fromtimestamp(current_ts).strftime('%Y-%m-%d-%H:%M:%S')
```

Comment:
No need to import `time` and then to `datetime`. It is possible to get the now time from datetime directly

time stamps with colon for filename might not be the best idea.

```python
>>> import datetime
>>> datetime.datetime.now()
datetime.datetime(2016, 8, 18, 11, 39, 30, 422410)
>>> datetime.datetime.now().isoformat()
'2016-08-18T11:39:36.208304'
>>> datetime.datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
'2016-08-18-11:40:47'
```

I would do instead with the `datetime` module

```python
>>> datetime.datetime.now().strftime('%Y-%m-%dT%H-%M-%S')
'2016-08-18T11-41-58'
```

Or if you really prefer the `time` module

```python
>>> time.strftime('%Y-%m-%dT%H-%M-%S', time.gmtime())
'2016-08-18T02-45-25'
```

The only difference is the TZ of the computer. Local or UTC time.